mmdet/datasets/custom.py

```diff
@@ -322,3 +322,34 @@ def evaluate(self,
                 for i, num in enumerate(proposal_nums):
                     eval_results[f'AR@{num}'] = ar[i]
         return eval_results
+
+    def __repr__(self):
+        """Print the number of instance number."""
+        dataset_type = 'Test' if self.test_mode else 'Train'
+        result = (f'\n{self.__class__.__name__} {dataset_type} dataset '
+                  f'with number of images {len(self)}, '
+                  f'and instance counts: \n')
+        if self.CLASSES is None:
+            return result
+        instance_count = np.zeros(len(self.CLASSES) + 1).astype(int)
+        # count the instance number in each image
+        for idx in range(len(self)):
+            label = self.get_cat_ids(idx)
+            unique, counts = np.unique(label, return_counts=True)
+            if len(unique) > 0:
+                # add the occurrence number to each class
+                instance_count[unique] += counts
+            else:
+                # background is the last index
+                instance_count[-1] += 1
+
+        for cls, count in enumerate(instance_count):
+            if cls < len(self.CLASSES):
+                result += f'[{self.CLASSES[cls]}]: {count}'
```

Comment:
![image](https://user-images.githubusercontent.com/17425982/112432670-2b7cea80-8d7c-11eb-96ca-e0449b007aa2.png)
This print format is not pretty

Reply:
Suggest to use tables to list them out, like that in the classwise coco mAP.

Reply:
reformat it into a table like coco mAP